src/core/types/arbitrageloops/mempoolLoop.ts

```diff
@@ -112,19 +189,79 @@ export class MempoolLoop {
 			if (mempoolTrades.length === 0) {
 				continue;
 			} else {
+				try{
 				applyMempoolTradesOnPools(this.pools, mempoolTrades);
+				} catch(e){
+					console.log("error in mempooltradesonpool")
+				}
 			}
 
-			const arbTrade = this.arbitrageFunction(this.paths, this.botConfig);
+			const arbTrade = this.arbitrageFunction(this.paths, this.botConfig, this.errorpaths);
 
 			if (arbTrade) {
-				await this.trade(arbTrade);
+				try {
+					await this.trade(arbTrade);
+				} catch (e) {
+					await this.errHandle(e);
+				}
 				arbTrade.path.cooldown = true;
 				break;
 			}
 		}
 	}
 
+
+	
+	/**
+	 * ERRORHANDLER 
+	 * TODO: Define other Error Codes.
+	 */
+	public async errHandle(err: any) {
```

Comment:
Not sure we want the error handling logic defined in the mempool loop. I could see this being used for a clob bot as well, that wont be using the same loop. Logic seems more high level than to a specific loop